spock/src/test/groovy/org/jboss/weld/spock/util/EmbeddedSpecRunnerWrapper.groovy

```diff
@@ -0,0 +1,24 @@
+package org.jboss.weld.spock.util
+
+import org.junit.platform.engine.DiscoverySelector
+import org.junit.platform.testkit.engine.EngineTestKit
+import spock.util.EmbeddedSpecRunner
+
+/**
+ * Work-around for a strange Groovy behavior.
+ *
+ * @author Björn Kautler
+ */
+class EmbeddedSpecRunnerWrapper extends EmbeddedSpecRunner {
```

Comment:
Why do we need this? I can see it is used in one test but I don't understand its purpose.

Reply:
Honestly I don't know yet why other than it doesn't work without.
I use the embedded spec runner to test the exception if there is automagic mode combined with `@WeldSetup`.
Unlike JUnit 5 where I had to do the ugly reflection hackery, Spock properly supports these tests using the embedded spec runner.
But for whatever reason, without this wrapper that just has a plain copy of the private `doRunRequest` method in the parent class, we get a `NoMethodFoundException`.
I'm in contact with one of the Groovy guys to look into this to maybe find out what the problem is and where to properly fix it.
But in the meantime this hack at least works.

Reply:
> I'm in contact with one of the Groovy guys to look into this to maybe find out what the problem is and where to properly fix it.

Curious issue indeed.
Would it make sense to add some more javadoc to capture what's the issue? I am pretty sure nobody will remember the next time we hit it :)

Reply:
I hope Paul will find out what it is and we can get rid of it or know what the problem is and can properly describe it then.
If you think it is unnecessary and remove it, you will pretty fast see what the problem is. :-D
Actually I hoped Paul would find out what the issue is, before we finish this PR, but he was short on time to look into it. :-(
I added basically the information from my last comment to the JavaDoc now.